src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserSearchController.kt

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.digital.hmpps.manageusersapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.manageusersapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.manageusersapi.model.AuthSource
+import uk.gov.justice.digital.hmpps.manageusersapi.service.Status
+import uk.gov.justice.digital.hmpps.manageusersapi.service.UserSearchService
+
+@RestController("lUserSearchController")
+class UserSearchController(
+  private val userSearchService: UserSearchService,
+) {
+
+  @GetMapping("/users/search")
+  @Operation(
+    summary = "Search for users ",
+    description = """
+      Search for users in the Auth DB only who match on partial first name, surname, username or email and return a pageable result set. 
+      Optionally choose the authentication sources from any combination of auth, delius, nomis and azuread sources.
+      It will default to AuthSource.auth if the authSources parameter is omitted.
```

Comment:
One to discuss, but wondering whether we should carry AuthSource.auth forward into this API? i.e. would it be better for this API to expose it as AuthSource.external? and provide an internal mapping from that to AuthSource.auth when calling Auth?

Reply:
It won't let me comment on earlier change - very odd .
Yes - the mapNotNull should work - I have created a method mapNotNull....we can see it works as the api call that gets sent out to auth does not contain the additional parameters (in the controllerIntTest) - hope that's ok